src/App.tsx

```diff
@@ -1,61 +1,154 @@
+import { useState, FormEvent } from 'react';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  return usersFromServer.find((user) => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const allowedChars = `0123456789
```

Comment:
Consider using a RegExp pattern to validate the input instead of listing all allowed characters in a string.